### ISSUE TITLE
Get battery level

### ISSUE THREAD

**Issue by meduar** [EliasKotlyar/Xiaomi-Dafang-Hacks#1290]
My camera is connected to a battery I want to check how much battery is remaining.

Any advice?

Thanks,

**Comment by jmtatsch** [comment]
If your camera has a ldr replace it with a voltage measurement circuit.

**Comment by mauckc** [comment]
@jmtatsch you are suggesting to replace the ldr in the circuit with some voltage measurement device? 

> If your camera has a ldr replace it with a voltage measurement circuit.

**Comment by jmtatsch** [comment]
Exactly

**Comment by meduar** [comment]
Is there any easy wait without doing soldering?

**Comment by jmtatsch** [comment]
I don't think there is a way without soldering and deeper electronics know how.
The older Dafangs/Xiaofangs had ldrs, the newer dont have it populated.
The basic idea is that the camera measures the voltage drop at the light dependent resistor with a gpio configured to read an analog input. This is where you could measure your battery voltage instead. You would have to scale that voltage to be in the range the gpio can work with though.

**Comment by ZipperZ** [comment]
Even without any LDR populated (at least on Xiaofang) it is possible to add a voltage divider using unmodified pcb. Somewhere here I have posted the schematic that I have probed out of unpopulated components. (If there is a need I can search for it). I have tested it by adding my own LDR as a part of voltage divider. Just I am not sure about the ADC reference value used... 
BUT... You will need soldering here, unless you are going to trust placed wires on pads, glued with hot snot... (And even for this heat involving tool is needed).
